src/lib/repl_service_ctx.cpp

```diff
@@ -19,11 +19,20 @@ const std::string repl_service_ctx_grpc::id_to_str(int32_t const id) const {
 
 const std::optional< Result< peer_id_t > > repl_service_ctx_grpc::get_peer_id(destination_t const& dest) const {
     if (std::holds_alternative< peer_id_t >(dest)) return std::get< peer_id_t >(dest);
-    if (std::holds_alternative< role_regex >(dest)) {
-        if (!_server) {
-            LOGW("server not initialized");
-            return folly::makeUnexpected(nuraft::cmd_result_code::SERVER_NOT_FOUND);
+
+    if (!_server) {
+        LOGW("server not initialized");
+        return folly::makeUnexpected(nuraft::cmd_result_code::SERVER_NOT_FOUND);
+    }
+
+    if (std::holds_alternative< svr_id_t >(dest)) {
+        if (auto const id_str = id_to_str(std::get< svr_id_t >(dest)); !id_str.empty()) {
+            return boost::uuids::string_generator()(id_str);
```

Comment:
I am not very clear how this can get the uuid for corresponding originator's uuid. Is it because the way we generate server_id (take some portion of uuid), that it can be used to call this api to get its uuid?

Reply:
`nuraft::raft_server::get_srv(...)` which is used to implement `id_to_str` gets the pair which is `srv_id, endpoint`. Endpoint is an opaque string to nuraft, but we store the string version of the uuid; so we just need to convert it back to boost::uuid here.

Reply:
The `id_to_str()` method [defined here https://github.com/eBay/nuraft_mesg/blob/main/src/lib/repl_service_ctx.cpp#L15] returns the `std::string` form of the uuid from the endpoint field